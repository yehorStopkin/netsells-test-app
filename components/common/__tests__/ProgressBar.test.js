const { shallowMount } = require('@vue/test-utils');
const ProgressBar = require('@/components/common/ProgressBar').default;

describe('ProgressBar', () => {
    test('empty bar', () => {
        const wrapper = shallowMount(ProgressBar, { propsData: { progress: 0 } });

        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const emptyBar = wrapper.find('.js-progress-bar__empty');
        const filledBar = wrapper.find('.js-progress-bar__filled');

        expect(emptyBar.exists()).toBeTruthy();
        expect(filledBar.exists()).toBeTruthy();
    });
    
    test('half-filled bar', () => {
        const progress = 50;
        const wrapper = shallowMount(ProgressBar, { propsData: { progress } });

        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const filledBar = wrapper.find('.js-progress-bar__filled');

        expect(filledBar.attributes('style') === `width: ${ progress }%`);
    });

    test('full-filled bar', () => {
        const progress = 100;
        const wrapper = shallowMount(ProgressBar, { propsData: { progress } });

        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();

        const filledBar = wrapper.find('.js-progress-bar__filled');

        expect(filledBar.attributes('style') === `width: ${ progress }%`);
    });
});
