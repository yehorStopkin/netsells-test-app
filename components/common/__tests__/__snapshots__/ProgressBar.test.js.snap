// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar empty bar 1`] = `
<div
  class="progress-bar"
>
  <div
    class="progress-bar__empty js-progress-bar__empty"
  />
   
  <div
    class="progress-bar__filled js-progress-bar__filled"
    style="width: 0%;"
  />
</div>
`;

exports[`ProgressBar full-filled bar 1`] = `
<div
  class="progress-bar"
>
  <div
    class="progress-bar__empty js-progress-bar__empty"
  />
   
  <div
    class="progress-bar__filled js-progress-bar__filled"
    style="width: 100%;"
  />
</div>
`;

exports[`ProgressBar half-filled bar 1`] = `
<div
  class="progress-bar"
>
  <div
    class="progress-bar__empty js-progress-bar__empty"
  />
   
  <div
    class="progress-bar__filled js-progress-bar__filled"
    style="width: 50%;"
  />
</div>
`;
